### PR TITLE
update flake

### PR DESCRIPTION
**Pull request by AndreyMlashkin**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,6 +10,8 @@ include(GNUInstallDirs)
 option(BUILD_DOCS "Build documentation" FALSE)
 option(ENABLE_TESTS "Enable tests" FALSE)
 
+find_package(Boost REQUIRED COMPONENTS container random filesystem log log_setup program_options thread system unit_test_framework timer)
```
Comment (akokoshn) [inline CMakeLists.txt:13]: why we need move it here?

File: run_tests.sh [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ninja executables_tests && \
```
Comment (akokoshn) [inline run_tests.sh:1]: why we can't use cmake with ENABLE_TESTS?

File: tests/libs/assigner_runner/runner_test.cpp [hunk 3]
```diff
@@ -9,33 +9,33 @@
...
-
-    // Checks below are disabled due to frequent changes in filling assignment tables
-    /*
-    auto assignments = runner.get_assignments();
-    ASSERT_EQ(assignments.size(), 2);
-    ASSERT_EQ(assignments[1].witness(0, 0), 8);
-    ASSERT_EQ(assignments[1].witness(0, 1), 4);
-    */
-}
+// TEST(runner_test, check_block) {
```
Comment (akokoshn) [inline tests/libs/assigner_runner/runner_test.cpp:12]: what happens with this test?

**Review by akokoshn — CHANGES_REQUESTED** [review]

File: run_tests.sh (hunk 2 above)
Comment (AndreyMlashkin) [reply run_tests.sh:1]: Not sure, what do you mean here?...

File: tests/libs/assigner_runner/runner_test.cpp (hunk 3 above)
Comment (AndreyMlashkin) [reply tests/libs/assigner_runner/runner_test.cpp:12]: It's failing with an assert

```
assigner: /nix/store/KYVWRGZ8V26L8MSHB6LYZL6DP52C7HY4-evm-assigner/include/rw.hpp:93: nil::evm_assigner::rw_operation<BlueprintFieldType> nil::evm_assigner::stack_operation(std::size_t, uint16_t, std::size_t, bool, zkevm_word<BlueprintFieldType>) [with BlueprintFieldType = nil::crypto3::algebra::fields::pallas_base_field; std::size_t = long unsigned int; uint16_t = short unsigned int]: Assertion `address < 1024' failed.
Aborted (core dumped)
```

File: CMakeLists.txt (hunk 1 above)
Comment (AndreyMlashkin) [reply CMakeLists.txt:13]: Without it, the configure phase is failing:

```
zkevm-framework> CMake Generate step failed.  Build files cannot be regenerated correctly.
error: builder for '/nix/store/vvvam9v0w98sb8aabvf2vhv616h4h2kz-zkevm-framework.drv' failed with exit code 1;
       last 25 log lines:
       > -- Configuring done (1.1s)
       > CMake Error at tests/libs/assigner_runner/CMakeLists.txt:3 (target_link_libraries):
       >   Target "assigner_runner_test" links to:
       >
       >     Boost::random
       >
       >   but the target was not found.  Possible reasons include:
       >
       >     * There is a typo in the target name.
       >     * A find_package call is missing for an IMPORTED target.
       >     * An ALIAS target is missing.
```

File: run_tests.sh (hunk 2 above)
Comment (AndreyMlashkin) [reply run_tests.sh:1]: Those tests are designed to be a cmake target. I am not changing this behaviour in this PR, it was like this before

File: CMakeLists.txt (hunk 1 above)
Comment (akokoshn) [reply CMakeLists.txt:13]: we have to fix it carefully.
Different libraries/binaries has different boost dependencies, so need allow way for partial build (only one target) without using useless packages

File: run_tests.sh (hunk 2 above)
Comment (akokoshn) [reply run_tests.sh:1]: before we don't use separate script for run tests

File: tests/libs/assigner_runner/runner_test.cpp (hunk 3 above)
Comment (akokoshn) [reply tests/libs/assigner_runner/runner_test.cpp:12]: so need to fix it

File: run_tests.sh (hunk 2 above)
Comment (AndreyMlashkin) [reply run_tests.sh:1]: Yes, right. I could put it back into flake file. Konstantin says, it's a good practice to have it in a separate script, so you can just call it from nix develop, i.e. it's reusable

File: tests/libs/assigner_runner/runner_test.cpp (hunk 3 above)
Comment (AndreyMlashkin) [reply tests/libs/assigner_runner/runner_test.cpp:12]: Yes, here is an issue for this already

File: CMakeLists.txt (hunk 1 above)
Comment (AndreyMlashkin) [reply CMakeLists.txt:13]: We are propagating the boost dependency now, so it's avoided

File: flake.nix [hunk 4]
```diff
@@ -20,29 +20,15 @@
...
-      rev = "3bd5b8df2091274abaa28fd86b9e3e89d661b95a";
-      inputs = {
-        nixpkgs.follows = "nixpkgs";
-        flake-utils.follows = "flake-utils";
-        nix-3rdparty.follows = "nix-3rdparty";
-      };
-    };
-    nil-zkllvm-blueprint = {
-      url = "https://github.com/NilFoundation/zkllvm-blueprint";
+      url = "https://github.com/NilFoundation/crypto3";
```
Comment (akokoshn) [inline flake.nix:26]: what reason switch to git type

File: zkevm-framework.nix [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-ggdb" else "")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-O0" else "")
+      "-G Ninja"
+  ];
+
+  doBuild = true;
+  doCheck = runTests;
+
+  checkPhase = ''
+    bash ../run_tests.sh
```
Comment (akokoshn) [inline zkevm-framework.nix:51]: please run tests by cmake

**Review by akokoshn — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 4 above)
Comment (AndreyMlashkin) [reply flake.nix:26]: It's just a copy-paste from zkllvm. Do you have a strong preference?

File: zkevm-framework.nix (hunk 5 above)
Comment (AndreyMlashkin) [reply zkevm-framework.nix]: done

**Review by akokoshn — APPROVED** [review]

File: zkevm-framework.nix [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  # enableDebugging will keep debug symbols in boost
+  propagatedBuildInputs = [ (if enableDebug then (enableDebugging boost_lib) else boost_lib) ];
+
+  buildInputs = [crypto3 evm-assigner intx ethash sszpp valijson gtest];
+
+  cmakeFlags =
+  [
+      (if runTests then "-DBUILD_TESTS=TRUE" else "")
+      (if runTests then "-DCMAKE_ENABLE_TESTS=TRUE" else "")
+      (if runTests then "-DENABLE_TESTS=TRUE" else "")
```
Comment (makxenov) [inline zkevm-framework.nix:40]: Why do we have so much flags to enable tests? I think it makes sense to create an issue at least, and unify all repositories to use the same flag. Currently it's very easy to forget about some flag and leave corresponding tests disabled.

File: zkevm-framework.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+      (if enableDebug then "-DCMAKE_BUILD_TYPE=Debug" else "-DCMAKE_BUILD_TYPE=Release")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-ggdb" else "")
+      (if enableDebug then "-DCMAKE_CXX_FLAGS=-O0" else "")
```
Comment (makxenov) [inline zkevm-framework.nix:43]: It should be a responsibility of `cmake` to set cpp flags, not of `nix`. I believe that `-O0` and `-g` will be set automatically with `-DCMAKE_BUILD_TYPE=DEBUG`. If you thing that `-ggdb` may be useful, please set it in `CMakeLists.txt`.

File: zkevm-framework.nix [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+  doCheck = runTests;
+
+  checkPhase = ''
+    ctest
+    ninja executables_tests
+  '';
+
+  shellHook = ''
+    PS1="\033[01;32m\]\u@\h\[\033[00m\]:\[\033[01;34m\]\w\[\033[00m\]\$ "
+    echo "Welcome to zkEVM framework development environment!"
```
Comment (makxenov) [inline zkevm-framework.nix:57]: Please print also build type, as it was in the original shell hook

File: zkevm-framework.nix (hunk 6 above)
Comment (akokoshn) [reply zkevm-framework.nix:40]: good catch, seems it was copy-past.
`ENABLE_TESTS` - enough

File: zkevm-framework.nix (hunk 8 above)
Comment (akokoshn) [reply zkevm-framework.nix]: done

File: zkevm-framework.nix (hunk 7 above)
Comment (akokoshn) [reply zkevm-framework.nix]: moved to cmake

**Review by makxenov — APPROVED** [review]

**Pull request merged**